docs/design-documents/tools/cmake.md

```diff
@@ -1,101 +1,118 @@
 # CMake Mbed OS
 
+Requirements:
+- CMake 3.13 and higher
+- mbed-tools (python 3.6 and higher)
+
 Two steps approach:
 
 - Mbed OS core CMake
-- Application CMake
+- Building an application with mbed-tools
 
-All what can be defined should be in CMake files and configuration (app/mbed .json files). Our build system would parse the configuration, create rules and generate top level CMake file that would include others (application CMake files plus also Mbed OS core CMake).
+Definitions and configurations would be defined in CMake files, Mbed app configuration file (`/path/to/app/mbed_app.json`) and Mbed library configuration files (`/path/to/mbed-os/<LIBRARY_NAME>/mbed_lib.json`). The Mbed build system would parse the Mbed configuration files and generate a `CMakeLists.txt` configuration file.
 
-To stay backward compatible, we create common rules as follows.
+The following rules must be respected for backward compatibility.
 
-We replace current `TOOLCHAIN_`, `FEATURE_`, `TARGET_` by providing `add_subdirectory` macros. There would be `if_target()`, `if_feature()`, `_if_toolchain()` macros for better readability.
+Target labels, components, and features defined in `/path/to/mbed-os/targets/targets.json` are used to help the build system determine which directories contain sources/include files to use in the build process. This is accomplished through a custom CMake function (`mbed_add_cmake_directory_if_labels`) to append a prefix (TARGET_, COMPONENT_ and FEATURE_) and add matching directories to the list of directories to process.
 
-Example, to add stm_directory only for STM targets (current rule TARGET_STM):
+An example, to add `TARGET_STM` in the folder `targets` where we have folders like TARGET_NXP, TARGET_STM, etc:
 
 ```
-add_subdirectory_if_target(STM stm_directory)
-
+mbed_add_cmake_directory_if_labels("TARGET")
 ```
 
-Application example:
+If a user selects for example target `NUCLEO_F411RE`, the target defines the label `STM`. As result, the target folder STM is included.
 
-```
-./TARGET_STM/qspi/driver_stm_qspi.cpp
-./TARGET_STM/peripheral_specific_to_stm/stm_specific_driver.cpp
-```
+The same could be applied to other labels like features or components.
 
-As result, the top level application CMake:
+To migrate to the new build system, we can provide auto scanning of the module and generate CMake based on what we find or use the way as described above. In both cases, we could stay backward compatible.
```

Comment:
I think we need to fix the grammar here, but I'm not sure what we mean by "auto scanning of the module"; do we mean "auto scanning of Mbed OS module and component directories"?

Reply:
File globbing is what I read

Reply:
I'll remove it, the migration path will come in separate update